devcareer/code-jammers-backend
# #175557405 Build features to allow an admin activate or deactivate a user

## What does this PR do ##
-  enables Admin to activate or deactivate a user

### Description of tasks to be completed? ###


### How should this be manually tested/checked? ###
- Clone the PR
- Checkout to the branch `ft-user-should-be-able-to-comment-175541724`
- Install the dependencies using `npm install`
- Create a `.env file` and define the following as follows:
   - `DEV_DATABASE_URL = <development database url>`
   - `TEST_DATABASE_URL = <test database url>`
   - `DATABASE_URL = <production database url>`
- run ```npm run migrate``` to create the database tables
- run ```npm run seed``` to seed the database
- Start the server by running ```npm run start-dev```
- Sign in via `http://localhost:3000/api/v1/users/signin` using a `POST` method with an admin details
```
     {
          "email": "ufuoma@gmail.com",
          "password": "12345"
     }
```
and copy the token (without the quotes) from the response body.

- Activate a user by making a patch request to `http://localhost:3000/api/v1/admin/activate-user/:id`
- Deactivate a user by making a patch request to `http://localhost:3000/api/v1/admin/deactivate-user/:id`
- where id is the id of the user to be activated or deactivated 

- Sign in via `http://localhost:3000/api/v1/users/signin` using a `POST` method with a deactivated user details
```
     {
          "email": "fiyin@gmail.com",
          "password": "12345"
     }
```
- it should return "Sorry User has been De-activated, Please contact an admin"

- Also try accessing any of the following routes with the de-activated user details
- update-profile patch route    `http://localhost:3000/api/v1/user-profile/`          ``

- recover password post route         `http://localhost:3000/api/v1/users/recover`

-  Should also return a  "Sorry User has been De-activated, Please contact an admin" error.

- Accessing the  comment route `http://localhost:3000/api/v1/comment`  without a token should return a ` "please login" ` error 


### Link to architecture
[architecture](https://docs.google.com/document/d/1leLB6Arb7uCCJewKytE-wZw_VgZM0pkYL50hbWaJpPw/edit)

### What are the relevant pivotal tracker stories?
[ #175557405](https://www.pivotaltracker.com/story/show/175557405)
